README.md

```diff
@@ -1,50 +1,153 @@
+### Docker Compose
 
-# Supported adapters
+To run the application using Docker Compose:
 
-## StandardCryptoPrice
+```bash
+docker-compose up
+```
 
-- This standard of adapter type is used for getting current price of each crypto symbols. Here are the spec of request, response, adapter input and adapter output.
-  - Request:
-    - symbols: str (e.g. "BAND,ALPHA")
-  - Response:
-    - prices: []{symbol: str, price: float, timestamp: integer} (e.g. [{symbol: “BAND”, price: 99999, timestamp: 161111111111}, {symbol: “ALPHA”, price: 99999, timestamp: 161111111111}])
-  - Adapter input:
-    - symbols: List[str] (e.g. ["BAND", "ALPHA"])
-  - Adapter output:
-    - prices: []{symbol: str, price: float, timestamp: integer} (e.g. [{symbol: “BAND”, price: 99999, timestamp: 161111111111}, {symbol: “ALPHA”, price: 99999, timestamp: 161111111111}]
+## Testing
 
-### CoinMarketCap
+Execute tests with:
 
-- To setup gateway for CoinmarketCap price API, you have to setup these environments:
-  - ADAPTER_TYPE = "standard_crypto_price"
-  - ADAPTER_NAME = "coin_market_cap"
-  - API_KEY = <YOUR_COIN_MARKET_CAP_API_KEY>
+```bash
+pytest
+```
 
-### CryptoCompare
+## Supported Adapters
 
-- To setup gateway for CryptoCompare price API, you have to setup these environments:
-  - ADAPTER_TYPE = "standard_crypto_price"
-  - ADAPTER_NAME = "crypto_compare"
-  - API_KEY = <YOUR_CRYPTO_COMPARE_API_KEY>
-  - 
-### CoinGecko
+### StandardCryptoPrice
 
-- To setup gateway for CoinGecko price API, you have to setup these environments:
-  - ADAPTER_TYPE = "standard_crypto_price"
-  - ADAPTER_NAME = "coin_gecko"
-  - API_KEY = <YOUR_COIN_GECKO_API_KEY>
+This adapter type retrieves the current price of each cryptocurrency symbol. Below are the specifications for the request, response, adapter input, and adapter output.
+
+#### GET Request
+
+Parameters:
+
+- `symbols`: str (e.g., "BAND,ALPHA")
+
+#### Response
+
+```python
+prices: []{symbol: str, price: float, timestamp: integer}
+```
+
+Example:
+
+```python
+[{symbol: "BAND", price: 99999, timestamp: 161111111111}, {symbol: "ALPHA", price: 99999, timestamp: 161111111111}]
+```
+
+#### Set Environment Valuables
+
+- CoinMarketCap:
+
+  - `ADAPTER_TYPE = "standard_crypto_price"`
+  - `ADAPTER_NAME = "coin_market_cap"`
+  - `API_KEY = <YOUR_COIN_MARKET_CAP_API_KEY>`
+
+- CryptoCompare:
+
+  - `ADAPTER_TYPE = "standard_crypto_price"`
+  - `ADAPTER_NAME = "crypto_compare"`
+  - `API_KEY = <YOUR_CRYPTO_COMPARE_API_KEY>`
+
+- CoinGecko:
+  - `ADAPTER_TYPE = "standard_crypto_price"`
+  - `ADAPTER_NAME = "coin_gecko"`
+  - `API_KEY = <YOUR_COIN_GECKO_API_KEY>`
+
+## VerifiableAi
+
+This VerifiableAi adapter type is used to request the Ai API.
+
+### OpenAI
+
+#### Request
+
+Body:
+
+```json
+{
+  "model": "gpt-3.5-turbo",
+  "messages": [
+    {
+      "role": "user",
+      "content": "What is the best French cheese?"
+    }
+  ],
+  "max_tokens": 25,
+  "seed": 1,
+  "stream": false,
+  "temperature": 0.7,
+  "top_p": 1
+}
```

Comment:
please help update Input and Request in verifiable AI to align with this actual request format as well.

Reply:
for both mistral and openAI